lib/puppet/type/cisco_itd_service.rb

```diff
@@ -0,0 +1,338 @@
+
+    munge { |value| value == 'default' ? :default : value.to_i }
+  end # property load_bal_method_end_port
+
+  newproperty(:load_bal_method_proto) do
+    desc 'ITD load balance protocol'
+
+    newvalues(:default, :tcp, :udp)
+  end # property load_bal_method_proto
+
+  newproperty(:load_bal_method_start_port) do
+    desc 'ITD load balance protocol port start range'
+
+    munge { |value| value == 'default' ? :default : value.to_i }
+  end # property load_bal_method_start_port
+
+  newproperty(:nat_destination) do
+    desc 'ITD Destination NAT'
+
+    newvalues(:true, :false, :default)
+  end # property nat_destination
+
+  newproperty(:peer_local) do
+    desc "ITD Peer involved in sandwich mode. Valid values are string,
+         ' keyword default'. "
+
+    munge do |value|
+      value = :default if value == 'default'
+      value
+    end
+  end # property peer_local
+
+  newproperty(:peer_vdc, array_matching: :all) do
+    format = '[vdc_name, service_name]'
+    desc 'An array of [vdc_name, service_name]'\
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
+  end # property peer_vdc
+
+  newproperty(:shutdown) do
+    desc 'ITD service shutdown'
+
+    newvalues(:true, :false, :default)
+  end # property shutdown
+
+  newproperty(:virtual_ip, array_matching: :all) do
+    format = '[virt_ip_str1, virt_ip_str2]'
+    desc 'An array of strings'\
+         "Valid values match format #{format}."
+
+    # Override puppet's insync method, which checks whether current value is
+    # equal to value specified in manifest.  Make sure puppet considers
+    # 2 arrays with same elements but in different order as equal.
+    def insync?(is)
+      (is.size == should.size && is.sort == should.sort)
+    end
+
+    def should_to_s(value)
+      value.inspect
+    end
+
+    def is_to_s(value)
+      value.inspect
+    end
+
+    munge do |value|
+      begin
+        return value = :default if value == 'default'
+        value
+      end
+    end
+  end # property virtual_ip
+
+  # Make sure that when nat_destination is in the manifest,
+  # next_hop is specified
+  def check_nat_ingress
+    return unless self[:nat_destination]
+    return if self[:ingress_interface][0] == :default
+    fail ArgumentError, 'ingress_interface not specified' if
+      self[:ingress_interface].nil?
+    self[:ingress_interface].each do |_intf, next_hop|
+      fail ArgumentError, 'next_hop not specified' if
```

Comment:
Can you add a bit more clarity to this error message so it's obvious that the `ingress_interface` property next_hop needs to be set?